modules/serverpod_auth/serverpod_auth_server/lib/src/business/sms_auth.dart

```diff
@@ -0,0 +1,156 @@
+import 'dart:math';
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_server/module.dart';
+import 'package:crypto/crypto.dart';
+import 'dart:convert';
+
+import 'package:serverpod_auth_server/src/business/user_images.dart';
+
+/// Collection of utility methods when working with SMS authentication.
+
+class SMSs {
+  /// Starts the process of authenticating a user with SMS.
+  /// Sends an SMS with an OTP to the given phone number And if user doesn't
+  /// exist, creates a new user with the given phone number after verifying the
+  /// OTP.
+  static Future<String> getOTPRequest({required String phoneNumber}) async {
+    assert(
+      AuthConfig.current.sendValidationEmail != null,
+      'The sendValidationEmail property needs to be set in AuthConfig.',
+    );
+
+    var otp = _generateOTP();
+
+    var hash = generateHashData(
+      phoneNumber: phoneNumber,
+      otp: otp,
+    );
+    await AuthConfig.current.sendValidationSMS!(
+      phoneNumber,
+      otp,
+    );
+    return hash;
+  }
+
+  /// Verifies that the OTP is correct and that the hash hasn't been tampered
+  /// with. If the OTP is correct, the user is logged in and a session is
+  /// returned.
+  static Future<bool> verifyOTPRequest({
+    required String phoneNumber,
+    required String otp,
+    required String storedHash,
+  }) async {
+    var verification = verifyHashData(
+      hashedData: storedHash,
+      phoneNumber: phoneNumber,
+      otp: otp,
+    );
+
+    if (!verification) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /// Creates a new user with the given phone number.
+  static Future<SmsAuth?> createUser(
+    Session session,
+    String phoneNumber,
+  ) async {
+    var userInfo = await Users.findUserByPhoneNumber(session, phoneNumber);
+
+    if (userInfo == null) {
+      userInfo = UserInfo(
+        userIdentifier: phoneNumber,
+        phoneNumber: phoneNumber,
+        userName: phoneNumber,
```

Comment:
Suggestion: Change this to an empty string.

UserName should be nullable. Since it is not required for the authentication endpoint we should indicate that it was never set by using an empty string.

Issue around nullable user names: https://github.com/serverpod/serverpod/issues/975